test/e2e/syncer/syncer_test.go

```diff
@@ -151,7 +121,12 @@ func TestSyncerLifecycle(t *testing.T) {
 			deployment, err = downstreamKubeClient.AppsV1().Deployments(downstreamNamespaceName).Get(ctx, upstreamDeployment.Name, metav1.GetOptions{})
 			require.NoError(t, err)
 			klog.Info(toYaml(deployment))
-			return expectedAvailableReplicas == deployment.Status.AvailableReplicas
+			if expectedAvailableReplicas == deployment.Status.AvailableReplicas {
+				return true
+			}
+			dumpPodEvents(downstreamKubeClient, downstreamNamespaceName, ctx, t)
```

Comment:
Curious: what were you debugging here?

Reply:
this is related to the previous discussion https://github.com/kcp-dev/kcp/pull/1127#discussion_r879707982 with @stevekuznetsov 

log the pod events in order to detect why the deployment has not the desired number of replicas, this way we don't have to wait for configmaps/tokens to be replicated and we can detect other situations where the deployment is failing.